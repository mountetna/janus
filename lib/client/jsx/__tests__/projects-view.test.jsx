import React from 'react';
import {Provider} from 'react-redux';
import { act, create } from 'react-test-renderer';
import {mockStore} from '../test-helpers';
import {stubUrl} from 'etna-js/spec/helpers';
import ProjectsView from '../projects-view';

describe('ProjectsView', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      user: {
	email: "janus@two-faces.org",
	name: "Janus Bifrons",
        permissions: { 
          administration: {
            privileged: false,
            project_name: 'administration',
            role: 'editor'
          }
        }
      }
    });
  });

  it('renders', async () => {
    const initialStubs = [
      stubUrl({
        verb: 'get',
	path: '/api/admin/projects',
        status: 200,
        response: {
          projects: [
            { project_name: "gateway", project_name_full: "Gateway"},
            { project_name: "tunnel", project_name_full: "Tunnel"},
            { project_name: "mirror", project_name_full: "Mirror"}
          ]
        }
      })
    ];

    let component = create(
      <Provider store={store}>
        <ProjectsView/>
      </Provider>
    );

    await act( async () => {
        await new Promise((resolve) => setTimeout(resolve, 15));
    });

    expect( component.toJSON() ).toMatchSnapshot()
  });
});
