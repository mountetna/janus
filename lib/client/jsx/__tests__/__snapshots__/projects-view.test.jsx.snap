// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsView renders 1`] = `
Array [
  <div
    id="admin-projects"
  >
    <div
      className="title"
    >
      Projects
    </div>
    <div
      className="MuiPaper-root MuiTableContainer-root makeStyles-table_container-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="all projects"
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Project Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Title
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Type
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </div>,
  <div
    id="new-project"
  >
    <div
      className="title"
    >
      New Project
    </div>
    <div
      className="item"
    >
      <div
        className="cell"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="project_name"
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="cell"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Project Title"
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="cell submit"
      >
        <span
          className="icon-grp icon-grp-approve  fa-stack fa-fw"
          onClick={[Function]}
        >
          <i
            className="icon approve  magic fa fa-2x fa-magic"
          />
        </span>
      </div>
    </div>
  </div>,
]
`;
